src/myuw-profile.js

```diff
@@ -1,198 +1,165 @@
-            and the menu will never open.
-        */
-        this.$button.addEventListener('click', e => {
-            e.stopPropagation();
-            this.$nav.classList.toggle('visible');
-
-            // Focus the menu upon opening, blur on close
-            if (this.$nav.classList.contains('visible')) {
-                this.$nav.focus();
-                this.$button.setAttribute('aria-expanded', 'true');
-            } else {
-                this.$nav.blur();
-                this.$button.setAttribute('aria-expanded', 'false');
-            }
-        });
-
-        // this.componentReady();
-
-        // Update the component to use the new attributes
-        this.updateAttribute('login-url');
-        this.updateAttribute('logout-url');
-        this.updateAttribute('background-color');
-    }
+  connectedCallback(){
+    this.eventListeners.forEach( ({target, type, handler}) => target.addEventListener(type, handler));
+  }
 
-    // Update the component with attribute values
-    updateAttribute(att) {
-      switch(att){
-        case 'login-url':
-          this.shadowRoot.getElementById('myuw-profile-login').setAttribute('href', this['login-url']);
-          break;
-
-        case 'logout-url':
-          this.shadowRoot.getElementById('myuw-profile-logout').setAttribute('href', this['logout-url']);
-          break;
-
-        case 'background-color':
-          this.$circle.style.backgroundColor = this['background-color'];
-          break;
-      }
-    }
+  disconnectedCallback(){
+    this.eventListeners.forEach( ({target, type, handler}) => target.removeEventListener(type, handler));
+  }
+
+  /**
+   * Add an on click event to profile bubble
+   * We need to make sure that we stop propagation on
+   * this event or else the window on click will always fire
+   * and the menu will never open.
+   */
+  handleButtonClick(event) {
+    event.stopPropagation();
+    this.toggleMenu();
+  }
 
-   /**
-    * Runs after component detects the 'myuw-login' event and receives
-    * the required parameter
-    * @param {*} person 
-    */
-    componentReady(person) {
-      if (person) {
-        if (person.firstName) {
-          // Add user's name to first menu item
-          this.$displayName.classList.remove('hidden');
-          this.$displayName.innerHTML = person.firstName;
-          // Change the letter in the profile circle
-          this.$circle.innerHTML = person.firstName.substring(0,1);
-          // Show the profile bubble
-          this.showProfileBubble();
-        } else {
-          this.$displayName.classList.add('hidden');
-          this.$circle.innerHTML = '<i class="material-icons">person</i>';
-          this.showProfileBubble();
-        }
-      } else {
-        this.showLoginButton();
-      }
+  handleKeydown(event) {
+    switch(event.key) {
+      default:
+        return; // let unhandled keys propogate
+      case 'Escape':
+        this.closeMenu();
+        break;
```

Comment:
I also tried to add keyboard up/down navigation, but I can't seem to actually focus the menuitems—I can call `focus()` on the right elements, but they just don't get focused and I don't know why!  Some say that the browser inspector itself messes up focus, but I tested with it closed too. :man_shrugging: 

```javascript
      case 'ArrowDown':
      case 'ArrowUp':
        const increment = event.key === 'ArrowUp'? 1 : -1;
        const menuItems = this.$nav.querySelectorAll('[role=menuitem]');
        console.log(menuItems); // FIXME: debug
        if (menuItems && menuItems.length) {
          const first = event.key === 'ArrowUp'? menuItems.length -1 : 0;
          const focused = Array.from(menuItems.values()).findIndex(menuItem => menuItem === document.activeElement);
          console.log(focused); // FIXME: debug
          if (focused === -1) {   
            if (menuItems[first]) {
              console.log('focus', menuItems[first]); // FIXME: debug
              menuItems[first].focus();
            }
          }
          else {
            if (menuItems[focused + increment]) {
              console.log('focus', menuItems[focused + increment]); // FIXME: debug
              menuItems[focused + increment].focus();
            }
            else {
              if (menuItems[first]) {
                console.log('focus', menuItems[first]); // FIXME: debug
                menuItems[first].focus();
              }
            }
          }
        }
        break;
```